### PR TITLE
Initial version of source rcon client.

### PR DESCRIPTION
**Pull request by stevenh**
Initial version of source rcon client featuring configurable multi-packet, supporting Counter-Strike Global Offensive, Minecraft and Starbound.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,56 @@
...
+Features
+--------
+* Full [Source RCON](https://developer.valvesoftware.com/wiki/Source_RCON_Protocol) Support.
+* [Multi-Packet Responses](https://developer.valvesoftware.com/wiki/Source_RCON_Protocol#Multiple-packet_Responses) Support.
+
+Supports
+--------
+* [Valve](http://www.valvesoftware.com/) [Counter-Strike Global Offensive](http://steamcommunity.com/app/730) and others.
+* [Mojang](https://mojang.com/) [Minecraft](https://minecraft.net/).
+* [Chuckelfish](https://chucklefish.org/) [Starbound](https://playstarbound.com/).
```
Comment (darkliquid) [inline README.md]: Mis-spelled Chucklefish here.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,56 @@
...
+package main
+
+import (
+	"log"
+
+        "github.com/multiplay/go-source"
+)
+
+func main() {
+        c, err := source.NewClient("192.168.1.102:27015", source.Password("mypass"))
```
Comment (darkliquid) [inline README.md]: Spurious leading space

File: client.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+// Client is a source rcon client.
+type Client struct {
+	conn    net.Conn
+	addr    string
+	pwd     string
+	timeout time.Duration
+	reader  *bufio.Reader
+	reqID   int32
+	read    func(expectedID int32) (string, error)
+	write   func(t int32, body string) error
```
Comment (darkliquid) [inline client.go]: t? maybe something a bit more informative?

File: client.go [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+	if err = c.auth(); err != nil {
+		c.conn.Close() // nolint: errcheck
+		return nil, err
+	}
+
+	return c, nil
+}
+
+// auth authenticates with the server if one was configured password.
```
Comment (darkliquid) [inline client.go]: if a password was set

File: client.go [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+				return "", err
+			}
+		case expectedID + 1:
+			// Response response packets, exactly two expected.
+			cnt++
+			switch cnt {
+			case 1:
+				// Echoed response packet.
+				if len(p.body) != 0 {
+					return "", ErrMalformedResponse("none empty body")
```
Comment (darkliquid) [inline client.go]: non-empty body

File: cmd.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+// WithArgs sets the command Args.
+func (c *Cmd) WithArgs(args ...interface{}) *Cmd {
+	c.args = args
+	return c
+}
+
+func (c *Cmd) String() string {
+	args := append([]interface{}{c.cmd}, c.args...)
+	// We use fmt.Sprintln + fmt.TrimSuffix as fmt.Sprintln garantees all
```
Comment (darkliquid) [inline cmd.go]: guarantees

**Review by milonoir — APPROVED** [review]

**Review by darkliquid — CHANGES_REQUESTED** [review]
Mostly minor typos and things

**Review by darkliquid — APPROVED** [review]

**Pull request merged**